src/RootCloseWrapper.js

```diff
@@ -26,7 +26,9 @@ export default class RootCloseWrapper extends React.Component {
     // If the click originated from within this component, don't do anything.
     // e.srcElement is required for IE8 as e.target is undefined
     let target = e.target || e.srcElement;
-    if (domUtils.contains(React.findDOMNode(this), target)) {
```

Comment:
I really don't think it's a good idea to introduce such a large change into master. It may not strictly be breaking, but it's a pretty big overhaul.